File: easy/max_sum_subarray/max_sum_subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_sum = -float("inf")
        current_sum = 0
        for n in nums:
            current_sum = max(n, current_sum+n)
            max_sum = max(max_sum, current_sum)

        return max_sum


s = Solution()
print("Solution 1 :", s.maxSubArray([-2, 1, -3, 4, -1, 2, 1, -5, 4]))

File: easy/array/max_sum_subarray/max_sum_subarray.go
package main

import (
	"fmt"
	"math"
)

func maxSubArray(nums []int) int {
	max_sum := math.MinInt64
	current_sum := 0

	for _, n := range nums {
		current_sum = int(math.Max(float64(n), float64(current_sum+n)))
		max_sum = int(math.Max(float64(max_sum), float64(current_sum)))
	}
	return max_sum
}

func main() {
	fmt.Println("Solution 1 : ", maxSubArray([]int{-2, 1, -3, 4, -1, 2, 1, -5, 4}))
}
